### PR TITLE
calls: 0.1.5 -> 0.2.0 with init of callaudiod and feedbackd

### PR DESCRIPTION
**Pull request by Pacman99**
###### Motivation for this change
To update calls and add its new dependencies: callaudiod and feedbackd. 

The feedbackd package is from @masipcat's #94820, I just bumped the version. I can remove it if that PR gets merged first, but its necessary for this version of calls to build right.

Also callaudiod and feedbackd should probably be setup as modules at some point, possibly as a part of #88767. I might try my hand at it sometime.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/callaudiod/default.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    owner = "mobian1";
+    repo = pname;
+    rev = version;
+    sha256 = "07k7xp5a9c4d4lq7amaj6cg6b3gsd77x9wvf7nzcf4vpaph4yiyj";
+  };
+
+  nativeBuildInputs = [
+    meson
+    ninja
+    pkgconfig
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/callaudiod/default.nix:26]:
```suggestion
    pkg-config
```

File: pkgs/applications/audio/callaudiod/default.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
+{ stdenv
+, fetchFromGitLab
+, meson
+, ninja
+, pkgconfig
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/callaudiod/default.nix:5]:
```suggestion
, pkg-config
```

File: pkgs/applications/misc/feedbackd/default.nix [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    gsound
+    libgudev
+    json-glib
+  ];
+
+  meta = with stdenv.lib; {
+    description = "A daemon to provide haptic (and later more) feedback on events";
+    homepage = "https://source.puri.sm/Librem5/feedbackd";
+    license = licenses.gpl3Plus;
+    maintainers = with maintainers; [ masipcat pacman99 ];
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/feedbackd/default.nix:47]: Please add yourself to maintainers.nix.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/feedbackd/default.nix (hunk 3 above)
Comment (Pacman99) [reply pkgs/applications/misc/feedbackd/default.nix:47]: I'm in the maintainer-list.nix, https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix#L6734

Comment (SuperSandro2000) [reply pkgs/applications/misc/feedbackd/default.nix]: Not sure why ofborg failed with this error. Maybe you where not at eval time.

Comment (Pacman99) [reply pkgs/applications/misc/feedbackd/default.nix]: Oh I just realized masipcat isn't yet on the list, I removed masipcat for now.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107681` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>callaudiod</li>
    <li>calls</li>
    <li>feedbackd</li>
  </ul>
</details>

File: pkgs/applications/misc/feedbackd/default.nix [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+, gsound
+, libgudev
+, json-glib
+, vala
+, gobject-introspection
+}:
+
+stdenv.mkDerivation rec {
+  pname = "feedbackd";
+  version = "0.0.0+git20201114";
```
Comment (davidak) [inline pkgs/applications/misc/feedbackd/default.nix:17]:
```suggestion
  version = "2020-11-14";
```

as written here https://nixos.org/manual/nixpkgs/stable/#sec-package-naming

Comment (davidak) [reply pkgs/applications/misc/feedbackd/default.nix:17]: i see that is their official release name. wtf. nevermind

**Comment by Pacman99** [comment]
Updated calls to 0.2.0 and fixed name and version for feedbackd to match the naming rules. Callaudiod cannot be updated, because calls depends on 0.4.0, this should change in the next calls version.

**Comment by davidak** [comment]
Result of `nixpkgs-review pr 107681` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>callaudiod</li>
    <li>calls</li>
    <li>feedbackd</li>
  </ul>
</details>

File: pkgs/applications/audio/callaudiod/default.nix [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  ];
+
+  buildInputs = [
+    alsaLib
+    libpulseaudio
+    glib
+  ];
+
+  meta = with stdenv.lib; {
+    description = "a daemon for dealing with audio routing during phone calls";
```
Comment (davidak) [inline pkgs/applications/audio/callaudiod/default.nix:36]:
```suggestion
    description = "Daemon for dealing with audio routing during phone calls";
```

meta.description should:

    Be capitalized.

that's what fedora uses and i think it makes sense to have the same description across distros, like the package names should also be the same

https://repology.org/project/callaudiod/packages

**Review by davidak — CHANGES_REQUESTED** [review]
one minor nitpick

**Comment by davidak** [comment]
Package builds and runs. Cant test the actual functionality.

Screenshot from 2021-01-21 22-18-07

One thing i noticed is that when i close it, the process still runs until i kill it with CTRL+C.

```
[nix-shell:~/.cache/nixpkgs-review/pr-107681]$ calls

(calls:27003): dbind-WARNING **: 22:18:50.186: Couldn't connect to accessibility bus: Failed to connect to socket /tmp/dbus-GXwHDL6PHu: Connection refused

(sm.puri.Calls:27003): Gtk-CRITICAL **: 22:18:50.276: gtk_application_uninhibit: assertion 'GTK_IS_APPLICATION (application)' failed

^C
```

**Review by davidak — APPROVED** [review]
LGTM

**Comment by Pacman99** [comment]
Yeah the description thing makes sense to me, just curious does nixpkgs have any specific guidelines for descriptions? I usually just copy directly from nixpkgs. 

>One thing i noticed is that when i close it, the process still runs until i kill it with CTRL+C.

I think thats intended, it is supposed to be also be a daemon, so it doesn't kill on application close. Its so it can receive phone calls.

**Comment by davidak** [comment]
> I usually just copy directly from upstream.

That makes sense most of the time. We have a few simple rules.

https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes

>Its so it can receive phone calls.

i see. good

File: pkgs/applications/audio/callaudiod/default.nix [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    pkg-config
+  ];
+
+  buildInputs = [
+    alsaLib
+    libpulseaudio
+    glib
+  ];
+
+  meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/callaudiod/default.nix:35]:
```suggestion
  meta = with lib; {
```
This got recently deprecated and shouldn't be used for new packages.

File: pkgs/applications/misc/feedbackd/default.nix [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+  ];
+
+  buildInputs = [
+    glib
+    gsound
+    libgudev
+    json-glib
+  ];
+
+  meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/feedbackd/default.nix:43]:
```suggestion
  meta = with lib; {
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 107681` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>callaudiod</li>
<li>calls</li>
<li>feedbackd</li>
</ul>
</details>

**Comment by Pacman99** [comment]
Updated feedbackd to 2021-01-25

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 107681` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>callaudiod</li>
<li>calls</li>
<li>feedbackd</li>
</ul>
</details>

**Comment by Pacman99** [comment]
Thanks for merging!

**Pull request merged**